File: Algorithms/024. Swap Nodes in Pairs/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None or head.next is None:
            return head
        else :
            ans=head.next
            head.next=self.swapPairs(ans.next)
            ans.next=head
            return ans

File: Algorithms/024. Swap Nodes in Pairs/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head==null||head.next==null) {
            return head;
        }else {
            int val=head.val;
            head.val=head.next.val;
            head.next.val=val;
            head.next.next=swapPairs(head.next.next);
        }
        return head;
    }
}